mongo/crud_examples_test.go

```diff
@@ -0,0 +1,128 @@
+	result, err := client.ListDatabases(context.TODO(), bson.D{{"name", "admin"}}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, db := range result.Databases {
+		fmt.Println(db.Name)
+	}
+}
+
+func ExampleClient_ListDatabaseNames() {
+	var client *mongo.Client
+
+	// use a filter to only select the admin database
+	result, err := client.ListDatabaseNames(context.TODO(), bson.D{{"name", "admin"}})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, db := range result {
+		fmt.Println(db)
+	}
+}
+
+func ExampleClient_Watch() {
+	var client *mongo.Client
+
+	// specify a pipeline that will add a field with the key "newField" to each event returned
+	// specify the MaxAwaitTimeOption to have each attempt wait two seconds for new documents
+	addFieldsStage := bson.D{{"$addFields", bson.D{{"newField", "this is a new field"}}}}
+	opts := options.ChangeStream().SetMaxAwaitTime(2 * time.Second)
+	changeStream, err := client.Watch(context.TODO(), mongo.Pipeline{addFieldsStage}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// print out all change stream events in the order they're received
+	// see the mongo.ChangeStream documentation for more examples of using change streams
+	for changeStream.Next(context.TODO()) {
+		fmt.Printf("got event %v\n", changeStream.Current)
+	}
+}
+
+func ExampleDatabase_Aggregate() {
+	var db *mongo.Database
+
+	// specify a pipeline that will list all sessions on the target server
+	// specify the MaxTime option to limit the amount of time the query can run on the server
+	localSessionsStage := bson.D{{"$listLocalSessions", bson.D{{"allUsers", true}}}}
+	opts := options.Aggregate().SetMaxTime(2 * time.Second)
+	cursor, err := db.Aggregate(context.TODO(), mongo.Pipeline{localSessionsStage}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// get a list of all returned documents and print them out
+	// see the mongo.Cursor documentation for more examples of using cursors
+	var results []bson.Raw
+	if err = cursor.All(context.TODO(), &results); err != nil {
+		log.Fatal(err)
+	}
+	for _, result := range results {
+		fmt.Println(result)
+	}
+}
+
+func ExampleDatabase_ListCollectionNames() {
+	var db *mongo.Database
+
+	// use a filter to only select the collection "foo"
+	result, err := db.ListCollectionNames(context.TODO(), bson.D{{"name", "foo"}})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, coll := range result {
+		fmt.Println(coll)
+	}
+}
+
+func ExampleDatabase_ListCollections() {
+	var db *mongo.Database
+
+	// use a filter to only select the collection "foo"
+	// specify the NameOnly option so the returned specificiations only have the name field populated
+	opts := options.ListCollections().SetNameOnly(true)
+	cursor, err := db.ListCollections(context.TODO(), bson.D{{"name", "foo"}}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// get a list of all returned specifications and print them out
+	// see the mongo.Cursor documentation for more examples of using cursors
+	var results []bson.Raw
```

Comment:
Nit: bson.M is more generally useful.  Consider using that type instead of bson.Raw

Reply:
done.